core/src/main/java/com/sopt/core/designsystem/component/dialog/NoostakDialog.kt

```diff
@@ -0,0 +1,154 @@
+import com.sopt.core.type.DialogType
+
+@Composable
+fun NoostakDialog(
+    dialogType: DialogType,
+    onClick: () -> Unit = {},
+    onDismissRequest: () -> Unit = {}
+) {
+    Dialog(
+        onDismissRequest = { onDismissRequest() },
+        properties = DialogProperties(
+            dismissOnBackPress = true,
+            dismissOnClickOutside = true
+        )
+    ) {
+        Card(
+            shape = RoundedCornerShape(dimensionResource(id = R.dimen.dialog_radius))
+        ) {
+            Column(
+                modifier = Modifier
+                    .width(274.dp)
+                    .wrapContentHeight()
+                    .background(
+                        color = NoostakTheme.colors.white
+                    ),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Spacer(
+                    modifier = Modifier.height(
+                        when (dialogType) {
+                            DialogType.GROUP -> 24.dp
+                            DialogType.LOGOUT -> 41.dp
+                            DialogType.WITHDRAWAL -> 30.dp
+                        }
+                    )
+                )
+                Text(
+                    text = stringResource(dialogType.content),
+                    textAlign = TextAlign.Center,
+                    style = if (dialogType == DialogType.GROUP) NoostakTheme.typography.c3Regular else NoostakTheme.typography.b4Regular
+                )
+                Spacer(
+                    modifier = Modifier.height(
+                        when (dialogType) {
+                            DialogType.GROUP -> 20.dp
+                            DialogType.LOGOUT -> 36.dp
+                            DialogType.WITHDRAWAL -> 26.dp
+                        }
+                    )
+                )
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(1.dp)
+                        .background(color = NoostakTheme.colors.gray200)
+                )
+                Row(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .height(IntrinsicSize.Min) // Row의 높이를 내부 컴포넌트에 맞춤
+                ) {
+                    Button(
+                        onClick = { onDismissRequest() },
+                        shape = RectangleShape,
+                        modifier = Modifier
+                            .weight(1f)
+                            .fillMaxHeight(),
+                        colors = ButtonDefaults.buttonColors(
+                            containerColor = NoostakTheme.colors.white,
+                            contentColor = NoostakTheme.colors.gray900,
+                            disabledContainerColor = NoostakTheme.colors.white,
+                            disabledContentColor = NoostakTheme.colors.gray900
+                        )
```

Comment:
P1: 모든 버튼에 리플 효과 없애기
```suggestion
                    Button(
                        onClick = { onDismissRequest() },
                        shape = RectangleShape,
                        modifier = Modifier
                            .weight(1f)
                            .fillMaxHeight(),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = NoostakTheme.colors.white,
                            contentColor = NoostakTheme.colors.gray900,
                            disabledContainerColor = NoostakTheme.colors.white,
                            disabledContentColor = NoostakTheme.colors.gray900
                        ),
                        interactionSource = NoRippleInteractionSource
```